File: 6. Sum Square Difference/solution.java
public class solution {
    public static void main(String[] args) {
        int n = 100;
        int difference = squareOfSum(n) - sumOfSquares(n);

        System.out.println(difference);
    }

    public static int sumOfSquares(int n) {
        int result = (n * (n + 1) * (2 * n + 1)) / 6;
        return result;
    }

    public static int squareOfSum(int n) {
        int result = ((n * (n + 1)) / 2);
        return result * result;
    }
}

File: 6. Sum Square Difference/solution.go
package main

import (
	"fmt"
	"math"
)

func main() {
	n := 100
	difference := squareOfSum(n) - sumOfSquares(n)

	fmt.Println(difference)
}

func sumOfSquares(n int) int {
	result := (n * (n + 1) * (2*n + 1)) / 6
	return int(math.Floor(float64(result)))
}

func squareOfSum(n int) int {
	result := int(math.Floor(float64(n*(n+1)) / 2))
	return result * result
}
